src/skagent/algos/maliar.py

```diff
@@ -253,6 +253,49 @@ def simulate_forward(
     return states_t_plus_1
 
 
+def _extract_network_parameters(network):
+    """Extract all parameters from a network into a flat tensor.
+
+    Helper function used by both training loops for convergence checking.
+
+    Parameters
+    ----------
+    network : torch.nn.Module
+        Neural network to extract parameters from
+
+    Returns
+    -------
+    torch.Tensor
+        Flattened parameter tensor
+    """
+    params = []
+    for param in network.parameters():
+        params.append(param.data.view(-1))
+    return torch.cat(params) if params else torch.tensor([])
+
+
+def _compute_parameter_difference(params1, params2):
```

Comment:
unclear why this is maliar method specific. utils.py, or ann.py? and why 'private' ? (_ preceding)